plugins/grid-bookmark/src/GridBookmarkWidget/components/Highlight/Highlight.tsx

```diff
@@ -0,0 +1,108 @@
+import { observer } from 'mobx-react'
+import { makeStyles } from 'tss-react/mui'
+import { SessionWithWidgets, getSession, notEmpty } from '@jbrowse/core/util'
+import { IExtendedLGV } from '../../model'
+
+// icons
+import BookmarkIcon from '@mui/icons-material/Bookmark'
+
+// locals
+import { Tooltip } from '@mui/material'
+import { GridBookmarkModel } from '../../model'
+
+type LGV = IExtendedLGV
+
+const useStyles = makeStyles()({
+  highlight: {
+    height: '100%',
+    position: 'absolute',
+    textAlign: 'center',
+    overflow: 'hidden',
+    display: 'flex',
+    alignItems: 'start',
+  },
+})
+
+const intensify = (rgba: string, opacity: number) => {
+  if (rgba) {
+    const values = rgba.replaceAll(/[^\d,.]/g, '').split(',')
+    const originalOpacity = values.pop()
+    // originalOpacity !== '0' assumes if the user has set the opacity of this highlight to 0, they probably don't want to see the label either
+    const n = `rgba(${values.join(', ')}, ${
+      originalOpacity !== '0' ? opacity : 0
+    })`
+    return n
+  }
+  return `rgba(0,0,0,0)`
+}
+
+const Highlight = observer(function Highlight({ model }: { model: LGV }) {
+  const { classes } = useStyles()
+
+  const session = getSession(model) as SessionWithWidgets
+  const { showBookmarkHighlights, showBookmarkLabels } = model
+  const assemblyNames = new Set(session.assemblyNames)
+
+  const bookmarkWidget = session.widgets.get(
+    'GridBookmark',
+  ) as GridBookmarkModel
+
+  const bookmarks = useRef(bookmarkWidget?.bookmarks ?? [])
+
+  useEffect(() => {
+    if (!bookmarkWidget) {
+      const newBookmarkWidget = session.addWidget(
+        'GridBookmarkWidget',
+        'GridBookmark',
+      ) as GridBookmarkModel
+      bookmarks.current = newBookmarkWidget.bookmarks
+    }
+  }, [session, bookmarkWidget])
+
+  return (
+    <>
+      {showBookmarkHighlights && bookmarks.current
+        ? bookmarks.current
+            .filter(value => assemblyNames.has(value.assemblyName))
+            .map(r => {
+              const s = model.bpToPx({
+                refName: r.refName,
+                coord: r.start,
+              })
+              const e = model.bpToPx({
+                refName: r.refName,
+                coord: r.end,
+              })
+              return s && e
+                ? {
+                    width: Math.max(Math.abs(e.offsetPx - s.offsetPx), 3),
+                    left: Math.min(s.offsetPx, e.offsetPx) - model.offsetPx,
+                    highlight: r.highlight,
+                    label: r.label,
+                  }
+                : undefined
+            })
+            .filter(notEmpty)
+            .map(({ left, width, highlight, label }, idx) => (
+              <div
+                key={`${left}_${width}_${idx}`}
+                className={classes.highlight}
+                style={{ left, width, background: highlight }}
+              >
+                {showBookmarkLabels ? (
```

Comment:
you could consider drawing the text right onto the display. this would make it easier to see at a glance, and allow screenshot-ability compared to requiring a mouseover

Reply:
do you mean have the label displayed at all times? is that desirable?

Reply:
IMO, yes.  the bookmark icon feels odd to me, especially if there is not any label. i think it may look better for there to be no icon, and text displayed always, unless showBookmarkLabels is false

Reply:
Not sure I agree with this --

You can turn the bookmark icon 'off' for the express reason of "I don't want this in my screenshots". Being enabled by default communicates to the user "what" the highlighted area is, and enables the label quick access. The icon, for this purpose, was requested by Lincoln when I first demo'd it to him (it looked like the first screenshot on this PR).

If a user really wants their labels in their screenshots, delayed screen captures exist. I think having so much text like that, especially for a heavy "annotator", would be obtuse.

Mostly though, I really don't see having the text displayed always as a better-looking option...I am having trouble conceiving what you might be imagining, here's a mockup?:
![image](https://github.com/GMOD/jbrowse-components/assets/83305007/f7151a92-e6f4-480d-8185-37e9cc9ec6af)
![image](https://github.com/GMOD/jbrowse-components/assets/83305007/31873596-7dfd-4945-ad87-bc0849e44873)

Maybe draft for me what you'd like to see or describe some more details.

Reply:
there are similar challenges with overlapping feature labels. I believe such overlapping text could be remedied in most cases by adding ellipses and/or perhaps adding y-axis dodging (so that they avoid being on the same text row)

i actually prefer the text being there despite overlaps, because it gives me a quick overview, without having to mouseover many things. it is up to you though

Reply:
I'll leave it as is for now...I think resolving the overlaps with an ellipses negates the idea of having an overview of the labels (because you can't read them in their entirety), and if you're looking for a big picture overview, you can see all your labels in the widget. Y-axis dodging could resolve for some cases, but I don't think it would scale well, because in higher density you're adding a lot of vertical space clutter.